plugins/entando-plugin-jpkiebpm/src/main/java/org/entando/entando/plugins/jpkiebpm/aps/system/services/bpmwidgetinfo/BpmWidgetInfoManager.java

```diff
@@ -154,29 +150,44 @@ public void setBpmWidgetInfoDAO(IBpmWidgetInfoDAO bpmWidgetInfoDAO) {
     protected IBpmWidgetInfoDAO getBpmWidgetInfoDAO() {
         return _bpmWidgetInfoDAO;
     }
+        
+    public IKieFormOverrideManager getKieFormOverrideManager() {
+        return _kieFormOverrideManager;
+    }
+
+    public void setKieFormOverrideManager(IKieFormOverrideManager kieFormOverrideManager) {
+        this._kieFormOverrideManager = kieFormOverrideManager;
+    }
 
     private IKeyGeneratorManager _keyGeneratorManager;
     private IBpmWidgetInfoDAO _bpmWidgetInfoDAO;
+    private IKieFormOverrideManager _kieFormOverrideManager;
 
     @Override
     public void updateFromPageChanged(PageChangedEvent event) {
         if (event.getOperationCode() == PageChangedEvent.REMOVE_OPERATION_CODE) {
-            this.getBpmWidgetInfoDAO().removeBpmWidgetInfo(event.getPage().getCode());
+            List<BpmWidgetInfo> bpmWidgetInfos = this.getBpmWidgetInfoDAO().searchBpmWidgetInfo(event.getPage().getCode());
+            if (null != bpmWidgetInfos) {
+                for (BpmWidgetInfo bpmWidgetInfo : bpmWidgetInfos) {
+                    this.removeBpmWidgetInfo(bpmWidgetInfo.getId());
+                }
+            }
         } else if (event.getOperationCode() == PageChangedEvent.EDIT_FRAME_OPERATION_CODE &&
                 event.getEventType().equals(PageChangedEvent.EVENT_TYPE_REMOVE_WIDGET)) {
             List<BpmWidgetInfo> bpmWidgetInfoList = this.getBpmWidgetInfoDAO().searchBpmWidgetInfo(event.getPage().getCode(), event.getFramePosition());
             if (null != bpmWidgetInfoList) {
 
                 for (final BpmWidgetInfo bpmWidgetInfo : bpmWidgetInfoList) {
-
-                    bpmWidgetInfo.setInformationDraft(null);
-                    bpmWidgetInfo.setFramePosDraft(null);
-                    this.getBpmWidgetInfoDAO().updateBpmWidgetInfo(bpmWidgetInfo);
+                    if(bpmWidgetInfo.getInformationOnline() == null) {
```

Comment:
Are we sure there is no false version of this? Always null?

Reply:
What to you mean by "no false version of this"?
If this variable is null it means that the widget is present only in the draft version of the page. So if it is removed we have to remove also the related widget information and form overrides.